### PR TITLE
Add Resolvers Implementation

### LINKED ISSUE #1525: Extract the Resolvers module, refactor it to implement the new contract and use the new RDF api

### PR DESCRIPTION
**Pull request by imsdu**
Fixes #1525

### CODE REVIEW

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala [hunk 1]
```diff
@@ -111,6 +114,15 @@ trait Resolvers {
...
     }
 
+  /**
+    * Lists all resolvers.
+    *
+    * @param pagination the pagination settings
+    * @param params     filter parameters for the listing
+    * @return a paginated results list
+    */
+  def list(pagination: FromPagination, params: ResolverSearchParams): UIO[UnscoredSearchResults[ResolverResource]]
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala:124]: At the API level, `Resolvers` are accessed in the scope of a project. So we probably need 2 things:
1. `ResolverSearchParams` should have a parameter which is `projectRef: ProjectRef`
2. 
```scala
def list(projectRef: ProjectRef, pagination: FromPagination, params: ResolverSearchParams): UIO[UnscoredSearchResults[ResolverResource]] - 
   list(pagination, params.copy( projectRef= Some(projectRef)))
```

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 2]
```diff
@@ -119,4 +120,15 @@ object SearchParams {
         organization.forall(_ == resource.value.organizationLabel)
   }
 
+  final case class ResolverSearchParams(
+      deprecated: Option[Boolean] = None,
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:124]: Add `projectRef: Option[ProjectRef]`

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverRejection._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverState.Initial
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination.FromPagination
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.ResultEntry.UnscoredResultEntry
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.ResolverSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.UnscoredSearchResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, IdSegment, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.UUIDF
+import ch.epfl.bluebrain.nexus.delta.sdk.{ResolverResource, Resolvers, _}
```
Comment (umbreak) [inline delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala:24]:
```scala
import ch.epfl.bluebrain.nexus.delta.sdk._
```

File: delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+      )
+    )
+
+  private def aggregate(config: AggregateConfig)(implicit as: ActorSystem[Nothing], clock: Clock[UIO]) = {
+    val definition = PersistentEventDefinition(
+      entityType = moduleType,
+      initialState = Initial,
+      next = Resolvers.next,
+      evaluate = Resolvers.evaluate,
+      tagger = (_: ResolverEvent) => Set(moduleType),
```
Comment (umbreak) [inline delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala:203]: If we want to support, as we discussed this morning:
- v1/resolvers/{org}/{project}/event
- v1/resolvers/{org}/event
- v1/resources/{org}/{project}/event (including resolvers)
- v1/resources/{org}/event (including resolvers)
...we will also need the following tags:
```scala
tagger = (ev: ResolverEvent) =>
        Set(
          moduleType,
          s"${Projects.moduleType}=${ev.project}",
          s"${Organizations.moduleType}=${ev.project.organization}"
        )
```

**Review by umbreak — CHANGES_REQUESTED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala (hunk 2 above)
Comment (bogdanromanx) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:124]: I don't think this is needed for the API; maybe just internally.

**Review by umbreak — APPROVED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala (hunk 2 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:135]: I don't think there is any harm. Alternatively we need another method on `ResolverSearchParams`:
```scala
def matches(resource: ResourceF[Resolver], projectRef: ProjectRef): Boolean
```

but I think it is more a hussle.

**Pull request merged**